chain/default_syncer_test.go

```diff
@@ -178,6 +178,40 @@ func requireSetTestChain(t *testing.T, con consensus.Protocol, mockStateRoots bo
 	}
 }
 
+// requireChainOfLength generates a fake chain of given length for testing
+// relating to chain length
+func requireChainOfLength(t *testing.T, length int, con consensus.Protocol) types.SortedCidSet {
```

Comment:
I would strongly prefer some way to avoid redundancy with [`requireGrowChain`](https://github.com/filecoin-project/go-filecoin/blob/master/chain/get_ancestors_test.go#L24).  I understand why you can't use that one here but a lot of the logic in both could be abstracted to something common.  Would you mind filing a follow-on / tech debt issue about building out a user-friendly chain growing framework for tests?